pkg/validate/openshift_cluster.go

```diff
@@ -0,0 +1,216 @@
+package validate
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/Azure/acs-engine/pkg/api/osa/vlabs"
+)
+
+var rxRfc1123 = regexp.MustCompile(`(?i)^` +
+	`([a-z0-9]|[a-z0-9][-a-z0-9]{0,61}[a-z0-9])` +
+	`(\.([a-z0-9]|[a-z0-9][-a-z0-9]{0,61}[a-z0-9]))*` +
+	`$`)
+
+var rxAgentPoolProfileVNetSubnetID = regexp.MustCompile(`(?i)^` +
+	`/subscriptions/[^/]+` +
+	`/resourceGroups/[^/]+` +
+	`/providers/Microsoft\.Network` +
+	`/virtualNetworks/[^/]+` +
+	`/subnets/[^/]+` +
+	`$`)
+
+var validAgentPoolProfileNames = map[string]struct{}{
+	string(vlabs.AgentPoolProfileRoleCompute): struct{}{},
+	string(vlabs.AgentPoolProfileRoleInfra):   struct{}{},
+	string(vlabs.AgentPoolProfileRoleMaster):  struct{}{},
+}
+
+var validRouterProfileNames = map[string]struct{}{
+	"default": struct{}{},
+}
+
+func isValidHostname(h string) bool {
+	return len(h) <= 255 && rxRfc1123.MatchString(h)
+}
+
+// OpenShiftCluster validates an OpenShiftCluster struct
+func OpenShiftCluster(oc *vlabs.OpenShiftCluster) (errs []error) {
+	if oc.Location == "" {
+		errs = append(errs, fmt.Errorf("invalid location %q", oc.Location))
+	}
+
+	if oc.Name == "" {
+		errs = append(errs, fmt.Errorf("invalid name %q", oc.Name))
+	}
+
+	if oc.Properties != nil {
+		errs = append(errs, validateProperties(oc.Properties)...)
+	}
+
+	return
+}
+
+func validateProperties(p *vlabs.Properties) (errs []error) {
```

Comment:
follow up: most of these errors don't give any suggestions on how to fix them (invalid formats don't show the format, invalid values don't show the acceptable set, etc)

Reply:
going to open a follow-up from this.  I am expecting specific guidance on error formatting from MSFT hence I haven't been to worried about this up to now.

Reply:
#127